storage/reads/string_iterator_writer.go

```diff
@@ -38,6 +38,10 @@ func (w *StringIteratorWriter) WrittenN() int {
 }
 
 func (w *StringIteratorWriter) WriteStringIterator(si cursors.StringIterator) error {
+	if si == nil {
```

Comment:
This should never be true, right? Just a precaution?

Reply:
right - only precautionary